cluster-autoscaler/cloudprovider/azure/azure_manager.go

```diff
@@ -643,21 +254,11 @@ func (m *AzureManager) getFilteredAutoscalingGroups(filter []labelAutoDiscoveryC
 		return nil, nil
 	}
 
-	switch m.config.VMType {
-	case vmTypeVMSS:
-		asgs, err = m.listScaleSets(filter)
-	case vmTypeStandard:
-		asgs, err = m.listAgentPools(filter)
-	case vmTypeAKS:
-		return nil, nil
-	default:
-		err = fmt.Errorf("vmType %q not supported", m.config.VMType)
-	}
-	if err != nil {
-		return nil, err
+	if m.config.VMType == vmTypeVMSS {
+		return m.listScaleSets(filter)
 	}
```

Comment:
@feiskyer I wanted to highlight this diff - was intentional to have the VM standard type as a valid type for autodisocery?

It seems that the `filter` was never used in listAgentPools so how do the min and max get populated? That's the only logic change in this PR - basically error out when trying to use autodiscovery for a non-supported `VMType`

Reply:
It should be a bug, only vmss support autodisocery.